File: src/main/java/me/lunaluna/fabric/elytrarecast/Startup.java
package me.lunaluna.fabric.elytrarecast;

import me.lunaluna.fabric.elytrarecast.config.Config;
import me.lunaluna.fabric.elytrarecast.config.ConfigHelper;
import net.fabricmc.api.ClientModInitializer;

import java.io.IOException;
import java.nio.file.Path;

public class Startup implements ClientModInitializer {
    public static Config config = null;
    public static Path configPath = Path.of("config", "elytra-recast.json");
    @Override
    public void onInitializeClient() {
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        try {
            config = ConfigHelper.getOrCreate(configPath);
        } catch (Exception e) {
            e.printStackTrace();
        }
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
        System.out.println("HELLO WORLD");
    }
}

File: src/main/java/me/lunaluna/fabric/elytrarecast/Startup.kt
package me.lunaluna.fabric.elytrarecast

import fi.dy.masa.malilib.config.ConfigManager
import fi.dy.masa.malilib.event.InitializationHandler
import fi.dy.masa.malilib.event.InputEventHandler
import fi.dy.masa.malilib.gui.GuiBase
import me.lunaluna.fabric.elytrarecast.config.UserConfig
import me.lunaluna.fabric.elytrarecast.config.gui.GuiUserConfig
import me.lunaluna.fabric.elytrarecast.hotkeys.InputHandler
import net.fabricmc.api.ClientModInitializer

class Startup : ClientModInitializer {
    override fun onInitializeClient() = InitializationHandler.getInstance().registerInitializationHandler {
        ConfigManager.getInstance().registerConfigHandler(Reference.MOD_ID, UserConfig)
        InputEventHandler.getKeybindManager().registerKeybindProvider(InputHandler)

        UserConfig.Hotkeys.OPEN_CONFIG.keybind.setCallback { _, _ ->
            GuiBase.openGui(GuiUserConfig())
            true
        }
        UserConfig.Hotkeys.TOGGLE_RECAST.keybind.setCallback { _, _ ->
            UserConfig.Recasting.ENABLED.booleanValue = !UserConfig.Recasting.ENABLED.booleanValue
            true
        }
        UserConfig.Hotkeys.TOGGLE_JUMP.keybind.setCallback { _, _ ->
            UserConfig.Jumping.ENABLED.booleanValue = !UserConfig.Jumping.ENABLED.booleanValue
            true
        }
    }
}